Sources/MongoKitten/AggregateStage.swift

```diff
@@ -1,227 +1,277 @@
+///   - localField: the name of the field in the input collection that shall match the `foreignField` in the `from` collection
+///   - foreignField: the name of the field in the `fromCollection` that shall match the `localField` in the input collection
+///   - newName: the collecting matches will be inserted as an array to the input collection, named as `newName`
+/// - Returns: an `AggregateBuilderStage`
+public struct Lookup: AggregateBuilderStage {
+    public internal(set) var stage: Document
+    public internal(set) var minimalVersionRequired: WireVersion? = nil
     
-	/// The `lookup` aggregation performs a join from another collection in the same database. This aggregation will add a new array to
-	/// your document including the matching documents.
-	///
-	/// # MongoDB-Documentation:
-	/// [Link to the MongoDB-Documentation](https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/)
-	///
-	/// # Example:
-	/// There are two collections, named `users` and `userCategories`. In the `users` collection there is a reference to the _id
-	/// of the `userCategories`, because every user belongs to a category.
-	///
-	/// If you now want to aggregate all users and the corresponding user category, you can use the `$lookup` like this:
-	///
-	/// ```
-	/// let pipeline = userCollection.aggregate([
-	///     .lookup(from: "userCategories", "localField": "categoryID", "foreignField": "_id", newName: "userCategory")
-	/// ])
-	///
-	/// pipeline.execute().whenComplete { result in
-	///    ...
-	/// }
-	/// ```
-	///
-	/// # Hint:
-	/// Because the matched documents will be inserted as an array no matter if there is only one item or more, you may want to unwind the joined documents:
-	///
-	/// ```
-	/// let pipeline = myCollection.aggregate([
-	///     .lookup(from: ..., newName: "newName"),
-	///     .unwind(fieldPath: "$newName")
-	/// ])
-	/// ```
-	///
-	/// - Parameters:
-	///   - from: the foreign collection, where the documents will be looked up
-	///   - localField: the name of the field in the input collection that shall match the `foreignField` in the `from` collection
-	///   - foreignField: the name of the field in the `fromCollection` that shall match the `localField` in the input collection
-	///   - newName: the collecting matches will be inserted as an array to the input collection, named as `newName`
-	/// - Returns: an `AggregateBuilderStage`
-    public static func lookup(
-        from: String,
-        localField: String,
-        foreignField: String,
-        as newName: String
-    ) -> AggregateBuilderStage {
-        return AggregateBuilderStage(document: [
-            "$lookup": [
-                "from": from,
-                "localField": localField,
-                "foreignField": foreignField,
-                "as": newName
-            ]
-        ])
+    public init(from: String, localField: String, foreignField: String, as newName: String) {
```

Comment:
This is great, but there's another initialiser that we need that includes a `pipeline` (as documented in the aggregate stage's reference).

https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/

The pipeline is an array of stages again, for which we can (and should) use AggregateBuilder.

Reply:
Also, we can use `FieldPath` for both `localField`, `foreignField` and `as`.

Reply:
When performing correlated subqueries, MongoDB requires you to specify the fields you want to use within the `let` field.

If you then want to reference any of these joined collection's fields within the pipeline stages, you are required to use $$ notation. This is a bit tricky since that requires the user to consider that with all the stages they want to include within the `pipeline`field.

Do you have any preferred solution for that?